colly.go

```diff
@@ -213,9 +217,12 @@ func (c *Collector) scrape(u, method string, depth int, requestData map[string]s
 	if len(c.requestCallbacks) > 0 {
 		c.handleOnRequest(request)
 	}
+
 	response, err := c.backend.Do(req, c.MaxBodySize)
-	// TODO add OnError callback to handle these cases
 	if err != nil {
+		if len(c.errorCallbacks) > 0 {
+			c.handleOnError(request, response, err)
```

Comment:
Perhaps, this should be also called if the response has non-200 status code. What do you think?

Reply:
Makes sense, thoughts on 301/302 redirects?

Reply:
Redirects are automatically handled by the http client IMO.

https://golang.org/pkg/net/http/#Client

Reply:
@tampajohn any update on this? I'd like to merge this PR asap because it is a very useful addition, so if you don't have time to code this, i can do it.